data/drupal/profile.ts

```diff
@@ -0,0 +1,394 @@
+      query ProfileSearch(
+        $queryByName: String = ""
+        $queryByResearchArea: String = ""
+        $page: Int = 0
+        $size: Int = 20
+        $units: [String] = []
+        $types: [String] = ""
+        $status: Boolean = null
+        $acceptingNewGrads: Boolean = null
+      ) {
+        profileSearch(
+          filter: {
+            queryByName: $queryByName
+            queryByResearchArea: $queryByResearchArea
+            units: $units
+            types: $types
+            status: $status
+            accepting_new_grads: $acceptingNewGrads
+          }
+          page: $page
+          pageSize: $size
+        ) {
+          results {
+            ...PartialProfile
+          }
+          pageInfo {
+            total
+          }
+        }
+      }
+    `),
+    variables: {
+      page: page,
+      size: pageSize,
+      units: units,
+      types: types,
+      queryByName: queryByName,
+      queryByResearchArea: queryByResearchArea,
+      status: showUnpublished ? null : true,
+      acceptingNewGrads: isAcceptingGraduateStudents,
+    },
+  });
+
+  if (error) {
+    handleGraphQLError(error);
+  }
+
+  if (!data) {
+    return { results: [], totalPages: 0 };
+  }
+
+  if (!data.profileSearch) {
+    return { results: [], totalPages: 0 };
+  }
+
+  let results: PartialProfileData[] = data.profileSearch.results;
+
+  if (showUnpublished) {
+    // The search database only indexes the current revision of a node, so we need to fetch the latest revision.
+    // Map each result to its latest revision.
+    // Unfortunately, this means making a request for each result individually.
+    // This is not ideal as it is bad for performance,
+    // but it is the best we can do for now,
+    // and since this will only be used in draft mode or the dev server, it should be fine.
+
+    const latestRevisionQuery = gql(/* gql */ `
+      query LatestRevisionProfile($id: ID!) {
+        nodeProfile(id: $id, revision: "latest") {
+          ...PartialProfile
+        }
+      }
+    `);
+
+    results = await Promise.all(
+      results.map(async (result) => {
+        const { data } = await client.query({
+          query: latestRevisionQuery,
+          variables: { id: result.id },
+        });
+
+        if (!data || !data.nodeProfile) {
+          console.warn(`Failed to fetch latest revision for profile ${result.id}, falling back to original result`);
+          return result;
+        }
+
+        return data.nodeProfile;
+      })
+    );
+  }
+
+  return {
+    results: results,
+    totalPages: Math.ceil(data.profileSearch.pageInfo.total / options.pageSize),
+    total: data.profileSearch.pageInfo.total,
+  };
+}
+
+export type ProfileType = NonNullable<ProfileTypeFragment>;
+
+export async function getAllTypes() {
+  const client = getClient();
+  const { data, error } = await client.query({
+    query: gql(/* gql */ `
+      query AllProfileTypes {
+        termProfileTypes(first: 4) {
```

Comment:
```suggestion
        termProfileTypes(first: 4, sortKey: WEIGHT) {
```

So we can move our preferred 4 (Faculty, Staff, etc.) at the top of the list on the backend